common/txmgr/inmemory_store.go

```diff
@@ -98,7 +100,31 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Creat
 	txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
 	error,
 ) {
-	return txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}, nil
+	tx := txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	if ms.chainID.String() != chainID.String() {
+		panic(fmt.Sprintf("create_transaction: invalid chain ID: %s", chainID))
+	}
+
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+	as, ok := ms.addressStates[txRequest.FromAddress]
+	if !ok {
+		return tx, fmt.Errorf("create_transaction: %w: %q", ErrAddressNotFound, txRequest.FromAddress)
+	}
```

Comment:
I don't think we should have this check here. We should instead initialize a new address state for the new `FromAddress`. Otherwise, a new key could have been added to the keystore that the Broadcaster is sending transactions with but we'd error on that in the txstore